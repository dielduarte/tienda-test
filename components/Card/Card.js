import styled from 'styled-components';
import { bool } from 'prop-types';

const Card = styled.div`
  background: ${props => props.theme.white};
  padding: 10px 20px;
  box-shadow: 0px 0px 1px 0px rgba(0, 0, 0, 0.25);
  border-radius: ${props => (props.bordered ? '3px' : 0)};
  margin-bottom: 20px;
  overflow-x: scroll;
`;

Card.propTypes = {
  bordered: bool
};

Card.defaultProps = {
  bordered: false
};

export default Card;
