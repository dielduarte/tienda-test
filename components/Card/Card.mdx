---
name: Card
---

import { Playground, PropsTable } from 'docz'
import Card from './Card';

# Card

<Playground>
  <Card>
    <h3>any children components here</h3>
  </Card>
</Playground>

## bordered

<Playground>
  <Card bordered>
    <h3>any children components here</h3>
  </Card>
</Playground>