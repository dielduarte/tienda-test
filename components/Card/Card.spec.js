import { shallow } from 'enzyme';
import 'jest-styled-components';

import Card from './Card';

describe('Card', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Card>
        <h1>title</h1>
      </Card>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly bordered', () => {
    const wrapper = shallow(
      <Card bordered>
        <h1>title</h1>
      </Card>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
