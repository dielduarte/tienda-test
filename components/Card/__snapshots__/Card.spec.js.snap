// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
.c0 {
  padding: 10px 20px;
  box-shadow: 0px 0px 1px 0px rgba(0,0,0,0.25);
  border-radius: 0;
  margin-bottom: 20px;
  overflow-x: scroll;
}

<div
  className="c0"
>
  <h1>
    title
  </h1>
</div>
`;

exports[`Card should render correctly bordered 1`] = `
.c0 {
  padding: 10px 20px;
  box-shadow: 0px 0px 1px 0px rgba(0,0,0,0.25);
  border-radius: 3px;
  margin-bottom: 20px;
  overflow-x: scroll;
}

<div
  className="c0"
>
  <h1>
    title
  </h1>
</div>
`;
